packages/ui/src/components/va-toast/VaToast.vue

```diff
@@ -34,148 +35,136 @@
-  color = prop<string>({ type: String, default: '' })
-  closeable = prop<boolean>({ type: Boolean, default: true })
-  onClose = prop<() => void>({ type: [Function, undefined] as any })
-  onClick = prop<() => void>({ type: [Function, undefined] as any })
-  multiLine = prop<boolean>({ type: Boolean, default: false })
-  position = prop<NotificationPosition>({
-    type: String as any,
-    default: 'top-right',
-    validator: (value: string) => {
-      return ['top-right', 'top-left', 'bottom-right', 'bottom-left'].includes(value)
-    },
-  })
-  render = prop<any>({ type: Function, default: undefined })
-}
-
-class VaToastRendererProps {
-  content = prop<any>({ type: Function, default: undefined })
-}
-
-const VaToastRendererPropsMixin = Vue.with(VaToastRendererProps)
-
-@Options({
+const VaToastRenderer = defineComponent({
   name: 'VaToastRenderer',
+  props: {
+    content: { type: Function, required: true },
+  },
+  setup: (props) => props.content(h),
 })
-class VaToastRenderer extends mixins(VaToastRendererPropsMixin) {
-  render () {
-    return this.content(h)
-  }
-}
-
-const ToastPropsMixin = Vue.with(ToastProps)
 
-@Options({
+export default defineComponent({
   name: 'VaToast',
   components: { VaIcon, VaToastRenderer },
   emits: ['on-click', 'on-close'],
-})
-export default class VaToast extends mixins(
-  ColorMixin,
-  ToastPropsMixin,
-) {
-  private timer: any = null
-  public visible = false
-
-  get positionX (): 'right' | 'left' {
-    return this.position.includes('right') ? 'right' : 'left'
-  }
+  props: {
+    title: { type: String as PropType<string>, default: '' },
+    offsetY: { type: Number as PropType<number>, default: 16 },
+    offsetX: { type: Number as PropType<number>, default: 16 },
+    message: { type: [String, Function], default: '' },
+    html: { type: Boolean as PropType<boolean>, default: false },
+    icon: { type: String as PropType<string>, default: 'close' },
+    customClass: { type: String as PropType<string>, default: '' },
+    duration: { type: Number as PropType<number>, default: 5000 },
+    color: { type: String as PropType<string>, default: '' },
+    closeable: { type: Boolean as PropType<boolean>, default: true },
+    onClose: { type: Function },
+    onClick: { type: Function },
+    multiLine: { type: Boolean as PropType<boolean>, default: false },
+    position: {
+      type: String as PropType<NotificationPosition>,
+      default: 'top-right',
+      validator: (value: string) => ['top-right', 'top-left', 'bottom-right', 'bottom-left'].includes(value),
+    },
+    render: { type: Function },
+  },
+  setup (props, { emit }) {
+    const { getColor } = useColors()
 
-  get positionY (): 'top' | 'bottom' {
-    return this.position.includes('top') ? 'top' : 'bottom'
-  }
+    const rootElement = ref<HTMLElement>()
 
-  get toastClasses () {
-    return [this.customClass]
-  }
+    const timer = ref<null | ReturnType<typeof setTimeout>>(null)
```

Comment:
This would be nice to have as composable. Timer also shouldn't be a reactive property.